compose: Make `/tmp` a directory by default, add `old-style-tmp`

cgwalters opened the pull request:
There are a few reasons to do this.  First, systemd changed to refuse
mounts on symlinks, and hence if one *wants* "/tmp-on-tmpfs", it's
annoying.

Second, the original rationale for having this symlink was that if you had
multiple ostree stateroots ("osnames"), it's nicer if they had the same `/tmp`
to avoid duplication. But in practice today that's already an issue due to
`/var/tmp`, and further the multiple-stateroot case is pretty unusual. And that
case is *further* broken by SELinux (if one wanted to have e.g. an Ubuntu and
Fedora) stateroots.  So let's fully decouple this and make `/tmp` a plain
old directory by default, so systemd's `tmp.mount` can become useful.

Now, things get interesting for the case where someone wants a physical `/tmp`
that *does* persist across reboots. Right now, if one just did a `systemctl mask
tmp.mount` as we do in Fedora Atomic Host's cloud images, you'd get a semantic
where `/tmp` stays per-deployment, which is weird.  Our recommendation for
that should likely be to set up a bind mount for `/tmp` → `/var/tmp`.

Closes: #669

cgwalters commented:
This one should probably wait until after the next rpm-ostree release - it's not urgent.  Perhaps we could flip the default and make it `new-style-tmp`?

jlebon commented:
Makes sense. If not doing this one soon, can you split out the first four commits in a separate PR?

> Or perhaps we could flip the default and make it `new-style-tmp`?

Yeah that sounds safer. (Maybe something more descriptive? E.g. `tmp-as-dir`?)

cgwalters commented:
Rebased and changed to `tmp-is-dir` (off by default).  We'll have to have a debate about continuing to `systemctl mask tmp.mount` in the cloud/vagrant boxes.

jlebon commented on tests/compose-tests/test-basic.sh:
```diff
@@ -38,3 +38,9 @@ echo "ok boot files"
+
+# https://github.com/projectatomic/rpm-ostree/issues/669
+ostree --repo=${repobuild} ls  ${treeref} /tmp > ls.txt
+assert_file_has_content ls.txt 'd01777 0 0      0 /tmp'
```
I think this one needs to be swapped for the symlink version now.

cgwalters commented:
This one should be good to go, tests just flaked.

jlebon commented:
@rh-atomic-bot r+ a2073f0
